contrib/mesos/pkg/scheduler/plugin.go

```diff
@@ -437,7 +437,7 @@ func (q *queuer) Run(done <-chan struct{}) {
 			}
 
 			pod := p.(*Pod)
-			if recoverAssignedSlave(pod.Pod) != "" {
+			if recoverAssignedSlave(pod.Pod) != "" || pod.InGracefulTermination() {
```

Comment:
Maybe we can simplify this one. The idea is that a pod is deleted before being scheduled (e.g. because no offer matched). I wonder who will delete such a pod though. Normally the kubelet has the logic to actually delete a pod after termination.